lib/block_view_validator.go

```diff
@@ -0,0 +1,2137 @@
+		DisableDelegatedStake:      validatorEntry.DisableDelegatedStake,
+		VotingPublicKey:            validatorEntry.VotingPublicKey.Copy(),
+		VotingPublicKeySignature:   validatorEntry.VotingPublicKeySignature.Copy(),
+		VotingSignatureBlockHeight: validatorEntry.VotingSignatureBlockHeight,
+		TotalStakeAmountNanos:      validatorEntry.TotalStakeAmountNanos.Clone(),
+		RegisteredAtBlockHeight:    validatorEntry.RegisteredAtBlockHeight,
+		LastActiveAtEpochNumber:    validatorEntry.LastActiveAtEpochNumber,
+		JailedAtEpochNumber:        validatorEntry.JailedAtEpochNumber,
+		ExtraData:                  copyExtraData(validatorEntry.ExtraData),
+		isDeleted:                  validatorEntry.isDeleted,
+	}
+}
+
+func (validatorEntry *ValidatorEntry) ToMapKey() ValidatorMapKey {
+	return ValidatorMapKey{
+		ValidatorPKID:           *validatorEntry.ValidatorPKID,
+		TotalStakeAmountNanos:   *validatorEntry.TotalStakeAmountNanos,
+		RegisteredAtBlockHeight: validatorEntry.RegisteredAtBlockHeight,
+		JailedAtEpochNumber:     validatorEntry.JailedAtEpochNumber,
+	}
+}
+
+func (validatorEntry *ValidatorEntry) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
+	var data []byte
+	data = append(data, EncodeToBytes(blockHeight, validatorEntry.ValidatorID, skipMetadata...)...)
+	data = append(data, EncodeToBytes(blockHeight, validatorEntry.ValidatorPKID, skipMetadata...)...)
+
+	// Domains
+	data = append(data, UintToBuf(uint64(len(validatorEntry.Domains)))...)
+	for _, domain := range validatorEntry.Domains {
+		data = append(data, EncodeByteArray(domain)...)
+	}
+
+	data = append(data, BoolToByte(validatorEntry.DisableDelegatedStake))
+	data = append(data, EncodeBLSPublicKey(validatorEntry.VotingPublicKey)...)
+	data = append(data, EncodeBLSSignature(validatorEntry.VotingPublicKeySignature)...)
+	data = append(data, UintToBuf(validatorEntry.VotingSignatureBlockHeight)...)
+	data = append(data, EncodeUint256(validatorEntry.TotalStakeAmountNanos)...)
+	data = append(data, UintToBuf(validatorEntry.RegisteredAtBlockHeight)...)
+	data = append(data, UintToBuf(validatorEntry.LastActiveAtEpochNumber)...)
+	data = append(data, UintToBuf(validatorEntry.JailedAtEpochNumber)...)
+	data = append(data, EncodeExtraData(validatorEntry.ExtraData)...)
+	return data
+}
+
+func (validatorEntry *ValidatorEntry) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
+	var err error
+
+	// ValidatorID
+	validatorID := &BlockHash{}
+	if exist, err := DecodeFromBytes(validatorID, rr); exist && err == nil {
+		validatorEntry.ValidatorID = validatorID
+	} else if err != nil {
+		return errors.Wrapf(err, "ValidatorEntry.Decode: Problem reading ValidatorID: ")
+	}
+
+	// ValidatorPKID
+	validatorPKID := &PKID{}
+	if exist, err := DecodeFromBytes(validatorPKID, rr); exist && err == nil {
+		validatorEntry.ValidatorPKID = validatorPKID
+	} else if err != nil {
+		return errors.Wrapf(err, "ValidatorEntry.Decode: Problem reading ValidatorPKID: ")
+	}
```

Comment:
Let's use the DecodeDeSoEncoders to clean this function up a bit.
```suggestion
	// ValidatorID
	validatorEntry.ValidatorID, err = DecodeDeSoEncoder(&BlockHash{}, rr)
	if err != nil {
		return errors.Wrapf(err, "ValidatorEntry.Decode: Problem reading ValidatorID: ")
	}

	// ValidatorPKID
	validatorEntry.ValidatorPKID, err = DecodeDeSoEncoder(&PKID{}, rr)
	if err != nil {
		return errors.Wrapf(err, "ValidatorEntry.Decode: Problem reading ValidatorPKID: ")
	}
```